File: pulsar-app/pulsar-examples/src/main/java/ai/platon/pulsar/examples/sites/amazon/AmazonCrawler.java
package ai.platon.pulsar.examples.sites.amazon;

import ai.platon.pulsar.context.PulsarContexts;

import java.util.Map;

class AmazonCrawler {

    public static void main(String[] argv) throws Exception {
        PulsarContexts.createSession().scrapeOutPages(
                "https://www.amazon.com/Best-Sellers/zgbs",
                "-outLink a[href~=/dp/]",
                Map.of("title", "#title", "ratings", "#acrCustomerReviewText")
        );
    }
}

File: pulsar-app/pulsar-examples/src/main/kotlin/ai/platon/pulsar/examples/sites/amazon/AmazonCrawler.kt
package ai.platon.pulsar.examples.sites.amazon

import ai.platon.pulsar.context.PulsarContexts

fun main() {
    val portalUrl = "https://www.amazon.com/Best-Sellers/zgbs"
    val args = "-i 1s -ii 5m -ol a[href~=/dp/] -ignoreFailure"

    val session = PulsarContexts.createSession()
    session.loadOutPages(portalUrl, args)
}
